src/components/UsernameValidationFormFields/index.tsx

```diff
@@ -82,19 +82,18 @@ const UsernameValidationForm = ({
 
   /**
    * renderButton
-   * Render the button to validate a username only if there's an input and
-   * if there's no success or error message.
+   * Render the button to validate a username and enable it only if
+   * the input is valid and if the input has been updated after it
+   * was checked for availability.
    */
   const renderButton = () => {
     const username = getValues("username");
-    const canValidate = inputValidation(username);
+    const canValidate =
+      inputValidation(username) && !usernameIsAvailable.message;
     return (
-      canValidate &&
-      !usernameIsAvailable.message && (
-        <Button onClick={validateUsername} type="button">
-          Check if username is available
-        </Button>
-      )
+      <Button onClick={validateUsername} type="button" disabled={!canValidate}>
```

Comment:
This now renders _all_ the time but the input now controls if it is disabled or not. This is to make it obvious how to re-check a value for availability.